setup/base/base.go

```diff
@@ -437,9 +434,27 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 					}
 				}
 			} else {
-				if err := externalServ.ListenAndServe(); err != nil {
-					if err != http.ErrServerClosed {
-						logrus.WithError(err).Fatal("failed to serve HTTP")
+				if externalHTTPAddr.IsUnixSocket() {
+					_ = os.Remove(externalHTTPAddr.Address)
```

Comment:
Needs docs to explain why you're removing this and why you're ignoring the return variable.

Reply:
fixed the ignore.
I think there is a general consensus about removing socket files on start, I think it is useful, but I have strong opinion.